src/core/grid_based_algorithms/lattice.cpp

```diff
@@ -45,17 +45,22 @@ int Lattice::init(double *agrid, double *offset, int halo_size, size_t dim) {
   }
 
   // sanity checks
+  bool incompatible_agrid = false;
   for (int dir = 0; dir < 3; dir++) {
     // check if local_box_l is compatible with lattice spacing
     if (fabs(local_box_l[dir] - this->grid[dir] * agrid[dir]) >
         epsilon * box_l[dir]) {
+      incompatible_agrid = true;
       runtimeErrorMsg() << "Lattice spacing agrid[" << dir << "]=" << agrid[dir]
                         << " is incompatible with local_box_l[" << dir
                         << "]=" << local_box_l[dir] << " ( box_l[" << dir
                         << "]=" << box_l[dir] << " node_grid[" << dir
                         << "]=" << node_grid[dir] << " )";
     }
   }
+  if (incompatible_agrid) {
+    return ES_ERROR;
```

Comment:
I think that does not help, since the return value is not checked... we could also directly return in after the ```runtimeErrorMsg```

Reply:
Thanks! Both the `.at(n)` solution and the return value check solution worked; we now get a clean `Exception: LB fluid activation` instead of the full MPI traceback.